lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1537,24 +1538,52 @@ namespace Js
             Assert(oldTypeToPromotedTypeMap && (Js::Var)oldTypeToPromotedTypeMap != scriptContext->GetLibrary()->GetUndefined());
             oldTypeToPromotedTypeMap = reinterpret_cast<TypeTransitionMap*>(oldTypeToPromotedTypeMap);
 
-            if (oldTypeToPromotedTypeMap->TryGetValue(reinterpret_cast<uintptr_t>(oldType), &cachedDynamicType))
+            RecyclerWeakReference<DynamicType>* weakRefCachedtype;
+            
+            if (oldTypeToPromotedTypeMap->TryGetValue(taggedOldType, &weakRefCachedtype))
             {
+                cachedDynamicType = weakRefCachedtype->Get();
```

Comment:
Can there ever be a case where cachedDynamicType got reclaimed, and then got reallocated, and we now go down this codepath thinking the type is there when in fact its not valid?